railgun/Makefile

```diff
@@ -104,6 +104,20 @@ endef
 
 # Kong Custom Targets
 
+# this will generate the custom typed clients needed for end-users implementing logic in Go to use our API types.
+# TODO: we're hacking around client-gen for now to enable it for enabled go modules, should probably contribute upstream to improve this.
+generate.clientsets:
+	@client-gen --go-header-file ./hack/boilerplate.go.txt \
+		--clientset-name clientset \
+		--input-base github.com/kong/kubernetes-ingress-controller/railgun/apis/  \
+		--input configuration/v1alpha1 \
+		--input-dirs github.com/kong/kubernetes-ingress-controller/railgun/apis/configuration/v1alpha1/ \
+		--output-base client-gen-tmp/ \
+		--output-package github.com/kong/kubernetes-ingress-controller/railgun/pkg/
```

Comment:
context: this weirdness is needed because it appears even the latest `client-gen` has issues when generating using module enabled repos.

`--output-package` is what `client-gen` will use as the base for import statements, so it MUST include the full import path, and `--output-base` would default to `$GOPATH` if not given, so this is a bit hacked to make it work without using `$GOPATH` shenanigans.

If you can think of a better way, please do feel free to commit it _straight to the branch_ I'm all for it, but where I left off it was looking like we need to make improvements to [client-gen](https://github.com/kubernetes/code-generator).